server/handlers.go

```diff
@@ -308,8 +308,9 @@ func killDeploymentHandler(w http.ResponseWriter, r *http.Request) {
 func listDeploymentsHandler(w http.ResponseWriter, r *http.Request) {
 	currentUser := getCurrentUser(r)
 	application := getCurrentApplication(r)
+	target := getTarget(application, r.URL.Query().Get("target"))
 
-	deployments, err := getAllApplicationDeployments(db, application)
+	deployments, err := getAllApplicationDeployments(db, application, target)
```

Comment:
I'd rather see a check here: if a filter-by-target-parameter is given then get only the target's deployments. Otherwise list all deployments. This way we could use the "old" `getAllApplicationDeployments` function and add a new one that filters by target.